### ISSUE TITLE
Broken Network Integration Test

### ISSUE THREAD

**Issue by syl22-00** [alexa/avs-device-sdk#523]
If I am not mistaken, your [Network Integration Test](https://github.com/alexa/avs-device-sdk/wiki/Unit-and-Integration-Tests#network-integration-test) has been broken for a while, it won't build.

* Commit 81c81b431ef84bf3a8d22a10d9bb1d2b8d7fb7a5 removed MessageRequest::Status, which is still used in `Integration/test/NetworkIntegrationTests.cpp`
* `PostConnectObject` is used in that file, but was not declared in the context.

The following changes seem to fix it:

```c++
diff --git a/Integration/test/NetworkIntegrationTests.cpp b/Integration/test/NetworkIntegrationTests.cpp
index cb96877..51faa3e 100644
--- a/Integration/test/NetworkIntegrationTests.cpp
+++ b/Integration/test/NetworkIntegrationTests.cpp
@@ -22,6 +22,7 @@
 #include <gtest/gtest.h>
 #include <ACL/AVSConnectionManager.h>
 #include <ACL/Transport/HTTP2MessageRouter.h>
+#include <ACL/Transport/PostConnectObject.h>
 #include <ContextManager/ContextManager.h>
 #include <AVSCommon/AVS/MessageRequest.h>
 #include <AuthDelegate/AuthDelegate.h>
@@ -91,7 +92,7 @@ public:
      */
     void sendEvent(
         const std::string& jsonContent,
-        MessageRequest::Status expectedStatus,
+        avsCommon::sdkInterfaces::MessageRequestObserverInterface::Status expectedStatus,
         std::chrono::seconds timeout,
         std::shared_ptr<avsCommon::avs::attachment::AttachmentReader> attachmentReader = nullptr);
 
@@ -168,7 +169,7 @@ void NetworkIntegrationTests::disconnect() {
 
 void NetworkIntegrationTests::sendEvent(
     const std::string& jsonContent,
-    MessageRequest::Status expectedStatus,
+    avsCommon::sdkInterfaces::MessageRequestObserverInterface::Status expectedStatus,
     std::chrono::seconds timeout,
     std::shared_ptr<avsCommon::avs::attachment::AttachmentReader> attachmentReader) {
     auto messageRequest = std::make_shared<ObservableMessageRequest>(jsonContent, attachmentReader);
@@ -228,7 +229,7 @@ TEST_F(NetworkIntegrationTests, testSendEventAfterDelayPass) {
     connect();
     addDelay(DELAY_TIME);
     sendEvent(
-        SYNCHRONIZE_STATE_JSON, MessageRequest::Status::SUCCESS, std::chrono::seconds(TIMEOUT_FOR_SEND_IN_SECONDS));
+        SYNCHRONIZE_STATE_JSON, avsCommon::sdkInterfaces::MessageRequestObserverInterface::Status::SUCCESS, std::chrono::seconds(TIMEOUT_FOR_SEND_IN_SECONDS));
     disconnect();
 }
 
@@ -241,7 +242,7 @@ TEST_F(NetworkIntegrationTests, testSendEventAfterDelayFails) {
     addDelay(LONG_DELAY_TIME);
     sendEvent(
         SYNCHRONIZE_STATE_JSON,
-        MessageRequest::Status::TIMEDOUT,
+        avsCommon::sdkInterfaces::MessageRequestObserverInterface::Status::TIMEDOUT,
         std::chrono::seconds(LONG_TIMEOUT_FOR_SEND_IN_SECONDS));
     disconnect();
 }
```

**Comment by mavamazon** [comment]
Thank you again! This kind of integration tests could be broken at the moment. We are going to work on this. 

I am not closing this ticket for you to be able to track the issue. If you feel that you are not interested in the exact moment of the fix, feel free to close it.

**Comment by syl22-00** [comment]
Thank you. It would be great to document which tests you know fail (or might fail). That would save people time trying to figure out if issues are on their end. For instance those tests always fail for me:

* AlexaDirectiveSequencerLibraryTest.sendDirectiveWithoutADialogRequestID
* AlexaDirectiveSequencerLibraryTest.oneBlockingDirectiveInTheMiddle
* AlexaCommunicationsLibraryTest.testSendEvent
* AlexaCommunicationsLibraryTest.testSendEventsConcurrently
* AlexaCommunicationsLibraryTest.testReceiveDirectiveOnDownchannel
* AlexaCommunicationsLibraryTest.testPersistentConnection
* AudioPlayerTest.FlashBriefing
* SpeechSynthesizerTest.handleMultipleConsecutiveSpeaks

**Comment by mavamazon** [comment]
Hmm... We usually disable those that are known to fail. Those you send does not seem to be the ones that should fail. Could you please provide more info about your environment?

**Comment by syl22-00** [comment]
Thanks, I opened another issue for that topic.

**Comment by mavamazon** [comment]
Ok, I assume that we can now close this ticket since you created a new one covering the problem in general. Feel free to reopen it if you have more questions regarding the original issue.

**Comment by syl22-00** [comment]
I made two tickets because I believe they are different: this one is about a part of the code that wont even build, the other ticket is about tests that do not pass.

**Comment by mavamazon** [comment]
I've reported a problem with Integration Tests for linux. Non-compiling code will also be covered while fixing the issues. However, feel free to reopen this ticket. Both of them will be reviewed and closed once the fix is ready.

**Comment by dhpp** [comment]
I believe the compilation issue has been resolved (NetworkIntegrationTests compiles for me on mac), and others have commented on the difference between our Unit and Integration tests - in particular, Unit tests should always pass, however Integration tests are more dependent on environment (including AVS itself), and so may be expected to fail on occasion.

I'll close this for now - please re-open if you believe your questions have not been fully addressed, thanks.
